source/write/bulk-write.txt

```diff
@@ -0,0 +1,287 @@
+update multiple other documents, then delete a document. Performing
+these operations by using their individual methods makes a call to the database
+for each method. You can instead use bulk write operations to perform
+these operations in a single database call.
+
+Performing Bulk Operations
+--------------------------
+
+To perform a bulk write operation, you must first create an instance of one of
+the following operation classes for each write operation you want to perform:
+
+- ``InsertOne``
+- ``UpdateOne``
+- ``UpdateMany``
+- ``ReplaceOne``
+- ``DeleteOne``
+- ``DeleteMany``
+
+Then, pass a list of these instances to the ``bulk_write()`` method.
+
+The following sections show how to create and use operation classes for
+different write operations.
+
+Insert Operations
+~~~~~~~~~~~~~~~~~
+
+To perform an insert operation, create an instance of ``InsertOne`` and specify
+the document you want to insert. To insert multiple documents, you must create
+an instance of ``InsertOne`` for each document.
+
+The following example creates an instance of ``InsertOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-insert-one
+   :end-before: end-bulk-insert-one
+   :language: python
+   :copyable:
+
+Update Operations
+~~~~~~~~~~~~~~~~~
+
+To perform a bulk update operation, create an instance of ``UpdateOne`` or
+``UpdateMany`` and pass it a **query filter** and the update operation you want
+to perform. A query filter specifies the criteria used to match documents to
+in your collection.
+
+``UpdateOne`` updates *the first* document that matches your query filter.
+``UpdateMany`` updates *all* documents that match your query filter.
+
+The following example creates an instance of ``UpdateOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-update-one
+   :end-before: end-bulk-update-one
+   :language: python
+   :copyable:
+
+The following example creates an instance of ``UpdateMany``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-update-many
+   :end-before: end-bulk-update-many
+   :language: python
+   :copyable:
+
+Replace Operations
+~~~~~~~~~~~~~~~~~~
+
+Replace operations remove all fields and values of a specified document and
+replaces them with new ones. To perform a replace operation, create an instance
+of ``ReplaceOne`` and pass it a query filter and the fields and values you want
+to replace with. To replace multiple documents, you must create
+an instance of ``ReplaceOne`` for each document.
+
+The following example creates an instance of ``ReplaceOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-replace-one
+   :end-before: end-bulk-replace-one
+   :language: python
+   :copyable:
+
+Delete Operations
+~~~~~~~~~~~~~~~~~
+
+To perform a delete operation, create an instance of ``DeleteOne`` or ``DeleteMany`` and pass it a
```

Comment:
S: Would probably follow a similar format here as for UpdateOne and UpdateMany above